src/chainstate/burn/operations/delegate_stx.rs

```diff
@@ -0,0 +1,290 @@
+    use crate::burnchains::bitcoin::address::{BitcoinAddress, BitcoinAddressType};
+    use crate::burnchains::bitcoin::{BitcoinInputType, BitcoinNetworkType, BitcoinTxInput, BitcoinTxOutput};
+    use crate::burnchains::{Txid, bitcoin::BitcoinTransaction};
+    use crate::chainstate::burn::Opcodes;
+    use crate::chainstate::burn::operations::DelegateStxOp;
+    use crate::types::chainstate::StacksAddress;
+    use crate::chainstate::stacks::address::PoxAddress;
+    use crate::chainstate::stacks::address::StacksAddressExtensions;
+    
+    #[test]
+    fn test_parse_delegate_stx() {
+        let tx = BitcoinTransaction {
+            txid: Txid([0; 32]),
+            vtxindex: 0,
+            opcode: Opcodes::DelegateStx as u8,
+            data: vec![1; 80],
+            data_amt: 0,
+            inputs: vec![BitcoinTxInput {
+                keys: vec![],
+                num_required: 0,
+                in_type: BitcoinInputType::Standard,
+                tx_ref: (Txid([0; 32]), 0),
+            }],
+            outputs: vec![
+                BitcoinTxOutput {
+                    units: 10,
+                    address: BitcoinAddress {
+                        addrtype: BitcoinAddressType::PublicKeyHash,
+                        network_id: BitcoinNetworkType::Mainnet,
+                        bytes: Hash160([1; 20]),
+                    },
+                },
+                BitcoinTxOutput {
+                    units: 10,
+                    address: BitcoinAddress {
+                        addrtype: BitcoinAddressType::PublicKeyHash,
+                        network_id: BitcoinNetworkType::Mainnet,
+                        bytes: Hash160([2; 20]),
+                    },
+                },
+                BitcoinTxOutput {
+                    units: 30,
+                    address: BitcoinAddress {
+                        addrtype: BitcoinAddressType::PublicKeyHash,
+                        network_id: BitcoinNetworkType::Mainnet,
+                        bytes: Hash160([0; 20]),
+                    },
+                },
+            ],
+        };
+
+        let sender = StacksAddress {
+            version: 0,
+            bytes: Hash160([0; 20]),
+        };
+        let op = DelegateStxOp::parse_from_tx(
+            16843022,
+            &BurnchainHeaderHash([0; 32]),
+            &BurnchainTransaction::Bitcoin(tx.clone()),
+            &sender,
+        )
+        .unwrap();
+
+        assert_eq!(&op.sender, &sender);
+        assert_eq!(
+            &op.reward_addr,
+            &Some(PoxAddress::Standard(
+                StacksAddress::from_bitcoin_address(&tx.outputs[1].address),
+                Some(AddressHashMode::SerializeP2PKH)
+            ))
+        );
+        assert_eq!(op.delegated_ustx, u128::from_be_bytes([1; 16]));
+        assert_eq!(op.delegate_to, StacksAddress::new(22, Hash160([1u8; 20])));
+        assert_eq!(op.until_burn_height, Some(u64::from_be_bytes([1; 8]))); 
```

Comment:
Can your test use different bit-patterns for the OP_RETURN fields, so we can be sure they're actually getting loaded correctly?

Also, can you add tests for malformed OP_RETURNs, as well as tests for when the `until_burn_height` is not set, as well as for both when the PoX address is and is not specified?

Reply:
I'm not exactly sure what you mean by malformed OP_RETURNs... Do you mean an incorrect opcode? Do you mean an malformed output?

Reply:
It also doesn't seem possible to construct an output that would trigger an error in the parsing code.